File: frontend-user/src/services/wishlist.service.ts
import axios from '../api/axios';

export const wishlistService = {
  getWishlist: async () => {
    const response = await axios.get('/wishlist');
    return response.data.data;
  },
  addToWishlist: async (bookId: number) => {
    const response = await axios.post(`/wishlist/add/${bookId}`);
    return response.data.data;
  },
  removeFromWishlist: async (bookId: number) => {
    const response = await axios.delete(`/wishlist/remove/${bookId}`);
    return response.data.data;
  },
  checkInWishlist: async (bookId: number) => {
    const response = await axios.get(`/wishlist/check/${bookId}`);
    return response.data.data;
  }
};

File: backend/src/main/java/com/fit/monolithic/backend/service/WishlistService.java
package com.fit.monolithic.backend.service;

import com.fit.monolithic.backend.dto.response.WishlistResponse;

public interface WishlistService {
    void addToWishlist(Long userId, Long bookId);
    void removeFromWishlist(Long userId, Long bookId);
    WishlistResponse getWishlist(Long userId);
    boolean isInWishlist(Long userId, Long bookId);
}
